packages/langium/src/lsp/completion/completion-provider.ts

```diff
@@ -239,6 +242,9 @@ export class DefaultCompletionProvider implements CompletionProvider {
     }
 
     protected completionForKeyword(keyword: ast.Keyword, context: AstNode | undefined, acceptor: CompletionAcceptor): MaybePromise<void> {
+        if (keyword.value.match(/^[\W]/) && keyword.value.length === 1 && !this.languageMetadata.showNonAlphabeticKeywords) {
```

Comment:
Hard to read... what is happening?

Reply:
Starts with a non word character of length 1?

Reply:
This checks if the the keyword start with a non-alphabetic token and is a single character.
I don't think hiding keywords such as `_keyword` from completion would be correct.

It also checks if the option `showNonAlphabeticKeywords ` is set to false.

For better readability, I guess we could first check for the option in an other if. What do you think?

Reply:
Yes, maybe a nested IF or a predicate function for the other two condition...
```js
function isNonAlphabeticToken() {
  return keyword.value.match(/^[\W]/) && keyword.value.length === 1;
}
```